pennylane/devices/qubit/apply_operation.py

```diff
@@ -228,9 +229,25 @@ def _(op: type_op, state):
         [1., 0.]], requires_grad=True)
 
     """
+    if op.has_sparse_matrix and not op.has_matrix:
+        return apply_operation_csr_matrix(op, state, is_state_batched)
```

Comment:
Potentially we could move this logic into `_apply_operation_default`? Since deferring to the sparse version is part of the "default" behavior.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/6883/commits/31bf7508b2a3ff9d714c1ef4836430500e72a9d7